requre/pytest_fixtures.py

```diff
@@ -0,0 +1,70 @@
+# MIT License
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import logging
+from pathlib import Path
+
+import pytest
+
+from requre.online_replacing import request_recording
+from requre.storage import PersistentObjectStorage
+from requre.utils import StorageMode, get_datafile_filename
+
+logger = logging.getLogger(__name__)
+
+
+def _get_storage_file(request) -> Path:
+    return get_datafile_filename(request.node)
+
+
+@pytest.fixture
+def record_requests_fixture(request):
+    storage_file = _get_storage_file(request)
+    with request_recording(storage_file=storage_file) as cassette:
+        mode_description = {
+            StorageMode.read: "replaying",
+            StorageMode.write: "recording",
+            StorageMode.append: "appending",
+            StorageMode.default: "in default mode",
+        }[PersistentObjectStorage().mode]
+        logger.debug(
+            f"Start requre {mode_description} with storage file: {storage_file}"
+        )
+        yield cassette
+        PersistentObjectStorage().dump()
+        logger.debug(f"End requre {mode_description} with storage file: {storage_file}")
+
+
+@pytest.fixture
+def remove_storage_file(request):
```

Comment:
what is reason for these two fixture functions? it is just for test purposes of requre? Because I do not see if it could be useful for other projects, it is not expected to remove files automatically, or I understand it in bad way?
in case it is just for purposes of requre, then I suppose to move it to tests modules.
or maybe more generic purpose is not to delete files, but just change storage file to `None` but in this case it has to be 
`PersistentObjectStorage().storage_file = None` and it will clear settings of PersistentStorage

Reply:
Mostly for the test purpose. I'll move them to the tests dir.

You cannot use that code directly in the test code since it is in the requre context.